src/Units/UnitState.java

```diff
@@ -157,48 +155,27 @@ public int alterHP(int change, boolean allowOver)
       // Apply the cap as needed
       final int newHP = Math.min(capHP, oldHP + change);
       // Figure out whether that reduces our healing
-      realChange = Math.min(change, newHP - oldHP);
+      realChange = Math.min(change, newHP - health);
     }
 
-    health = Math.max(1, health + healthFromHP(realChange));
+    health = Math.max(1, health + realChange);
     // Round HP up, if healing
-    if( change >= 0 )
-      health = healthFromHP(getHP());
+    if( roundUp && change >= 0 )
+      health = getHP();
```

Comment:
Strange to use `getHP` here instead of `roundHP`. Shrug.

Reply:
Same math either way, lol